src/lamp_py/ingestion/compress_gtfs/schedule_details.py

```diff
@@ -0,0 +1,346 @@
+import re
+import os
+import csv
+import zipfile
+import datetime
+import zoneinfo
+
+from urllib import request
+from typing import List
+from io import BytesIO
+from io import TextIOWrapper
+from dataclasses import dataclass
+from dataclasses import field
+
+import polars as pl
+
+from lamp_py.runtime_utils.process_logger import ProcessLogger
+
+from .gtfs_schema_map import gtfs_schema
+
+
+def file_as_bytes_buf(file: str) -> BytesIO:
+    """
+    Create buffer of local or http(s) file path
+
+    :return BytesIO buffer
+    """
+    if file.startswith("http"):
+        with request.urlopen(file) as response:
+            return BytesIO(response.read())
+
+    with open(file, "rb") as f:
+        return BytesIO(f.read())
+
+
+# pylint: disable=R0902
+# Too many instance attributes
+@dataclass
+class ScheduleDetails:
```

Comment:
i keep going back and forth on if this class should be expanded from a data class to include the functions in the `gtfs_to_parquet.py` file. they take a `ScheduleDetails` instance (or a reference to its members) as an argument. the idea of a class that represents a schedule that is able to generate a poldars dataframe that is then run through the `merge_frame_with_parquet` is appealing.

that said, this worked fine for me and i'm not sure what the advantages are other than maybe being a bit easier to read.

Reply:
My general goal was the keep the schedule operations separate from the parquet operations. After a few iterations, this current format seemed like the cleanliest delineation of those two things.